modules/gke-cluster/main.tf

```diff
@@ -48,7 +48,12 @@ resource "google_container_cluster" "cluster" {
   enable_autopilot = var.enable_features.autopilot ? true : null
 
   # the default nodepool is deleted here, use the gke-nodepool module instead
-  # node_config {}
+  node_config {
+    shielded_instance_config {
```

Comment:
sorry for doing this in pieces, a bit of a hectic day, we might want to use a dynamic block definition here and avoid the block if both flags are false, something like

```hcl
dynamic node_config {
  for_each = var.enable_features.shielded_nodes == true  [""] : []
  content {
      enable_secure_boot = true
      enable_integrity_monitoring = true
  }
}
```